### PR TITLE
Fixed some doxygen warnings

### PR DESCRIPTION
**Pull request by ahcorde**
Fixed some doxygen warnings

@ivanpauno and/or @hidmic do you mind to fix the rest of the doxygen warnings (as you are the main contributors of the repo) ? 

```bash
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/graph_cache.hpp:529: warning: Compound rmw_dds_common::EntityInfo is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/graph_cache.hpp:522: warning: Compound rmw_dds_common::ParticipantInfo is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/context.hpp:31: warning: Compound rmw_dds_common::Context is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/graph_cache.hpp:344: warning: Member DemangleFunctionT (typedef) of class rmw_dds_common::GraphCache is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/gid_utils.hpp:29: warning: Member operator()(const rmw_gid_t &lhs, const rmw_gid_t &rhs) const (function) of class rmw_dds_common::Compare_rmw_gid_t is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/context.hpp:32: warning: Member gid (variable) of class rmw_dds_common::Context is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/context.hpp:33: warning: Member pub (variable) of class rmw_dds_common::Context is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/context.hpp:34: warning: Member sub (variable) of class rmw_dds_common::Context is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/context.hpp:35: warning: Member graph_cache (variable) of class rmw_dds_common::Context is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/context.hpp:36: warning: Member node_update_mutex (variable) of class rmw_dds_common::Context is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/context.hpp:37: warning: Member listener_thread (variable) of class rmw_dds_common::Context is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/context.hpp:38: warning: Member thread_is_running (variable) of class rmw_dds_common::Context is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/context.hpp:39: warning: Member listener_thread_gc (variable) of class rmw_dds_common::Context is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/context.hpp:40: warning: Member graph_guard_condition (variable) of class rmw_dds_common::Context is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/graph_cache.hpp:539: warning: Member EntityInfo(const std::string &topic_name, const std::string &topic_type, const rmw_gid_t &participant_gid, const rmw_qos_profile_t &qos) (function) of class rmw_dds_common::EntityInfo is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/graph_cache.hpp:344: warning: Member DemangleFunctionT (typedef) of class rmw_dds_common::GraphCache is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/graph_cache.hpp:501: warning: Member NodeEntitiesInfoSeq (typedef) of class rmw_dds_common::GraphCache is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/graph_cache.hpp:503: warning: Member EntityGidToInfo (typedef) of class rmw_dds_common::GraphCache is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/graph_cache.hpp:504: warning: Member ParticipantToNodesMap (typedef) of class rmw_dds_common::GraphCache is not documented.
src/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/graph_cache.hpp:505: warning: Member GidSeq (typedef) of class rmw_dds_common::GraphCache is not documented.
Generating docs for compound rmw_dds_common::ParticipantInsrc/ros2/rmw_dds_common/rmw_dds_common/include/rmw_dds_common/graph_cache.hpp:523: warning: Member node_entities_info_seq (variable) of class rmw_dds_common::ParticipantInfo is not documented.
```

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

**Comment by ivanpauno** [comment]
> @ivanpauno and/or @hidmic do you mind to fix the rest of the doxygen warnings (as you are the main contributors of the repo) ?

will do

**Comment by ivanpauno** [comment]
@ahcorde I've fixed (almost) all the warnings.

There's still two related to the links between markdown files:
- README.md -> rmw_dds_common/QUALITY_DECLARATION.md
- rmw_dds_common/QUALITY_DECLARATION.md -> CONTRIBUTING.md

I don't know how to solve them.
I guess you're experiencing the same problem in other repos, so you'll have to figure it out.

**Comment by ahcorde** [comment]
I found in the doxygen repo an issue with relative links https://github.com/doxygen/doxygen/issues/7595

We can skip for now these two issues

Thank you!

File: rmw_dds_common/include/rmw_dds_common/context.hpp [hunk 1]
```diff
@@ -27,16 +27,27 @@
 namespace rmw_dds_common
 {
 
+/// Structure of data that a Context will need in any rmw implementation
```
Comment (hidmic) [inline rmw_dds_common/include/rmw_dds_common/context.hpp:30]: @ahcorde @ivanpauno nit:
```suggestion
/// Base data structure that a Context will need in any DDS-based RMW implementation
```

File: rmw_dds_common/include/rmw_dds_common/context.hpp [hunk 2]
```diff
@@ -27,16 +27,27 @@
...
+/// mapping one Participant to Multiple Nodes.
 struct Context
 {
+  /// Gid of the Participant that the context uses.
```
Comment (hidmic) [inline rmw_dds_common/include/rmw_dds_common/context.hpp:34]: @ahcorde @ivanpauno nit: 
```suggestion
  /// Global ID of the Participant that the Context uses.
```

File: rmw_dds_common/include/rmw_dds_common/context.hpp (hunk 1 above)
Comment (hidmic) [reply rmw_dds_common/include/rmw_dds_common/context.hpp:30]: I'd also mention that this implies one participant per context.

File: rmw_dds_common/include/rmw_dds_common/context.hpp [hunk 3]
```diff
@@ -27,16 +27,27 @@
...
   rmw_gid_t gid;
+  /// Publisher used to publish ParticipantEntitiesInfo discovery data.
   rmw_publisher_t * pub;
+  /// Subscriber used to listen ParticipantEntitiesInfo discovery data.
```
Comment (hidmic) [inline rmw_dds_common/include/rmw_dds_common/context.hpp:38]: @ivanpauno @ahcorde nit:
```suggestion
  /// Subscriber used to listen to ParticipantEntitiesInfo discovery data.
```

File: rmw_dds_common/include/rmw_dds_common/context.hpp [hunk 4]
```diff
@@ -27,16 +27,27 @@
...
   rmw_subscription_t * sub;
+  /// Cached graph from listened discovery data.
```
Comment (hidmic) [inline rmw_dds_common/include/rmw_dds_common/context.hpp:40]: @ahcorde @ivanpauno nit:
```suggestion
  /// Cached graph from discovery data.
```
as it includes in-process data, no?

File: rmw_dds_common/include/rmw_dds_common/context.hpp [hunk 5]
```diff
@@ -27,16 +27,27 @@
...
   GraphCache graph_cache;
+  /// Mutex that should be locked when updating graph cache/publishing a message.
```
Comment (hidmic) [inline rmw_dds_common/include/rmw_dds_common/context.hpp:42]: @ahcorde @ivanpauno nit:
```suggestion
  /// Mutex that should be locked when updating the graph cache...
```
I'm unsure as to how to rephrase. Publish what message?

File: rmw_dds_common/include/rmw_dds_common/context.hpp [hunk 6]
```diff
@@ -27,16 +27,27 @@
...
   std::mutex node_update_mutex;
+  /// Thread where to listen to discovery data.
   std::thread listener_thread;
+  /// Indicates if the thread is running.
```
Comment (hidmic) [inline rmw_dds_common/include/rmw_dds_common/context.hpp:46]: @ahcorde @ivanpauno nit: 
```suggestion
  /// Indicates if the listener thread is running.
```

File: rmw_dds_common/include/rmw_dds_common/context.hpp [hunk 7]
```diff
@@ -27,16 +27,27 @@
...
   rmw_gid_t gid;
+  /// Publisher used to publish ParticipantEntitiesInfo discovery data.
   rmw_publisher_t * pub;
+  /// Subscriber used to listen ParticipantEntitiesInfo discovery data.
   rmw_subscription_t * sub;
+  /// Cached graph from listened discovery data.
   GraphCache graph_cache;
+  /// Mutex that should be locked when updating graph cache/publishing a message.
   std::mutex node_update_mutex;
+  /// Thread where to listen to discovery data.
```
Comment (hidmic) [inline rmw_dds_common/include/rmw_dds_common/context.hpp:44]: @ahcorde @ivanpauno nit: 
```suggestion
  /// Thread to listen to discovery data.
```

File: rmw_dds_common/include/rmw_dds_common/graph_cache.hpp [hunk 8]
```diff
@@ -498,10 +499,18 @@ class GraphCache
    * @}
    */
 
+  /// Internal, do not use.
```
Comment (hidmic) [inline rmw_dds_common/include/rmw_dds_common/graph_cache.hpp:502]: @ahcorde @ivanpauno FYI, there is an [\internal](https://www.doxygen.nl/manual/commands.html#cmdinternal) Doxygen command.

File: rmw_dds_common/include/rmw_dds_common/graph_cache.hpp [hunk 9]
```diff
@@ -518,19 +527,28 @@ RMW_DDS_COMMON_PUBLIC
 std::ostream &
 operator<<(std::ostream & ostream, const GraphCache & topic_cache);
 
+/// Structure to represent the discovery data of a Participant.
 struct ParticipantInfo
 {
+  /// Discovery information of each Node created from a Participant.
   GraphCache::NodeEntitiesInfoSeq node_entities_info_seq;
+  /// Name of the enclave.
```
Comment (hidmic) [inline rmw_dds_common/include/rmw_dds_common/graph_cache.hpp:535]: @ahcorde @ivanpauno nit: perhaps we can link to ROS 2 security documentation?

File: rmw_dds_common/include/rmw_dds_common/context.hpp (hunk 1 above)
Comment (ivanpauno) [reply rmw_dds_common/include/rmw_dds_common/context.hpp]: > I'd also mention that this implies one participant per context.

That was already mentioned in the next line.

File: rmw_dds_common/include/rmw_dds_common/graph_cache.hpp (hunk 8 above)
Comment (ivanpauno) [reply rmw_dds_common/include/rmw_dds_common/graph_cache.hpp]: Done https://github.com/ros2/rmw_dds_common/pull/26/commits/462c77547848704b78218103ae068db6a666e728

File: rmw_dds_common/include/rmw_dds_common/context.hpp (hunk 5 above)
Comment (ivanpauno) [reply rmw_dds_common/include/rmw_dds_common/context.hpp]: rephrased, see https://github.com/ros2/rmw_dds_common/pull/26/commits/462c77547848704b78218103ae068db6a666e728

File: rmw_dds_common/include/rmw_dds_common/graph_cache.hpp (hunk 9 above)
Comment (ivanpauno) [reply rmw_dds_common/include/rmw_dds_common/graph_cache.hpp:535]: I would prefer not, links can easily get broken in the future.
The first result of googling "ros2 enclave" is the appropriate documentation.

**Comment by ahcorde** [comment]
friendly ping @hidmic

File: rmw_dds_common/include/rmw_dds_common/graph_cache.hpp (hunk 9 above)
Comment (hidmic) [reply rmw_dds_common/include/rmw_dds_common/graph_cache.hpp:535]: Fair enough.

**Comment by ivanpauno** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11745/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6825/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9582/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/11682/)

**Pull request merged**